ldap/src/main/java/org/springframework/security/ldap/DefaultSpringSecurityContextSource.java

```diff
@@ -74,7 +78,12 @@ public DefaultSpringSecurityContextSource(String providerUrl) {
 			rootDn = (rootDn != null) ? rootDn : urlRootDn;
 		}
 		setUrls(urls.toArray(new String[0]));
-		setBase(rootDn);
+		try {
+			setBase((rootDn != null) ? URLDecoder.decode(rootDn, StandardCharsets.UTF_8.toString()) : null);
```

Comment:
For added readability, consider moving the exception handling out to a separate method, e.g.

```java
private static String urlDecode(String text) {
    try {
        return URLDecoder.decode(text, StandardCharsets.UTF_8.toString());
    }
    catch (UnsupportedEncodingException ex) {
        return new IllegalStateException(ex);
    }
}
```